tests/mock_server.py

```diff
@@ -157,60 +217,89 @@ def json():
     "/cases/c58e7a75-e349-476a-9020-247af6b2e92b": MockResponseCase,
     "-python-client-v2": MockResponseVersions,
     "/account": MockResponseOrganizationAccounts,
+    "/folders/items/folder-3834758b-3d39-4a4a-ad85-710b7652267c/metadata": MockResponseFolderRootMetadata,
+    "/folders/items/folder-4da3cdd0-c5b6-4130-9ca1-196237322ab9/metadata": MockResponseFolderNestedMetadata,
+}
+
+PUT_RESPONSE_MAP = {
+    "/folders/move": MockResponseFolderSubmit,
+}
+
+POST_RESPONSE_MAP = {
+    "/volumemeshes/00112233-4455-6677-8899-aabbccddeeff/case": MockResponseCaseSubmit,
+    "/volumemeshes/00000000-0000-0000-0000-000000000000/case": MockResponseCaseSubmit,
+    "/folders": MockResponseFolderSubmit,
 }
 
 
-def mock_webapi(url, params):
+def mock_webapi(type, url, params):
     method = url.split("flow360")[-1]
     print("<><><><><><<><<><<><><>><<>")
 
-    print(method)
+    print(type, method)
+
+    if type == "get":
+        if method in GET_RESPONSE_MAP.keys():
+            return GET_RESPONSE_MAP[method]()
+
+        if method.startswith("-python-client-v2"):
+            return MockResponseVersions
 
-    if method in RESPONSE_MAP.keys():
-        return RESPONSE_MAP[method]()
+        if method.startswith("/volumemeshes/page"):
+            if params["includeDeleted"]:
+                return MockResponseVolumeMeshesPageWithDeleted()
+            return MockResponseVolumeMeshesPage()
 
-    if method.startswith("-python-client-v2"):
-        return MockResponseVersions
+        if method.startswith("/volumemeshes"):
+            if params["includeDeleted"]:
+                return MockResponseVolumeMeshesWithDeleted()
+            return MockResponseVolumeMeshes()
 
-    if method.startswith("/volumemeshes/page"):
-        if params["includeDeleted"]:
-            return MockResponseVolumeMeshesPageWithDeleted()
-        return MockResponseVolumeMeshesPage()
+        if method.endswith("/auth/credential"):
```

Comment:
Just as a note to self, as this is my code - I found no elegant way of mocking requests to the portal API endpoint, if more requests to the portal API appear in the future we should probably make a separate response map.